test/sol5/kyberNetwork.js

```diff
@@ -876,6 +881,32 @@ contract('KyberNetwork', function(accounts) {
             };
         });
 
+        describe("test gas helper", async() => {
+            before("setup, add and list reserves", async() => {
+                //init reserves
+                let result = await nwHelper.setupReserves(network, tokens, 3, 0, 0, 0, accounts, admin, operator);
+
+                reserveInstances = result.reserveInstances;
+                numReserves += result.numAddedReserves * 1;
+
+                //add and list pair for reserve
+                await nwHelper.addReservesToNetwork(network, reserveInstances, tokens, operator);
+            });
+
+            after("unlist and remove reserve", async() => {
+                await nwHelper.removeReservesFromNetwork(network, reserveInstances, tokens, operator);
+                reserveInstances = {};
+            });
+
+            it("test that gas helper can't revert trade even if it reverts", async() => {
+                platformFeeBps = new BN(50);
+                hint = await nwHelper.getHint(rateHelper, matchingEngine, reserveInstances, hintType, undefined, ethAddress, destToken.address, ethSrcQty);
+
+                await network.tradeWithHintAndFee(network.address, ethAddress, ethSrcQty, destToken.address, taker, 
```

Comment:
can you somehow show it reverted internally?
maybe print tx reciept and see if this data exists there.
not sure about it

Reply:
It's hard to see in tx receipt, but I was able to check the condition that will makes it revert. Also, checked by changing `address(gasHelper).call()...` to `gasHelper.freeGas()...` to confirm that it reverts.